ext/date/php_date.c

```diff
@@ -4357,7 +4357,11 @@ static int php_date_interval_initialize_from_hash(zval **return_value, php_inter
 		zval *z_arg = zend_hash_str_find(myht, element, sizeof(element) - 1); \
 		if (z_arg && Z_TYPE_P(z_arg) <= IS_STRING) { \
 			zend_string *str = zval_get_string(z_arg); \
-			DATE_A64I((*intobj)->diff->member, ZSTR_VAL(str)); \
+			if (ZSTR_LEN(str)) { \
+				DATE_A64I((*intobj)->diff->member, ZSTR_VAL(str)); \
+			} else { \
+				(*intobj)->diff->member = -99999; \
+			} \
```

Comment:
I think it would make more sense to treat `days` separately, rather than trying to fit it into this macro (and thus giving all other properties the same behavior, even though it doesn't make sense there).

We could then also explicitly check for a `false` value, rather than `"" === (string)false`, as what is done now.

Reply:
I've created a separate macro for handling `days` property.

But I'm not sure if it's the best way to solve this problem.